capa/capabilities/dynamic.py

```diff
@@ -65,14 +69,29 @@ def find_thread_capabilities(
     # might be found at different calls, that's ok.
     call_matches: MatchResults = collections.defaultdict(list)
 
+    # matches found at the sequence scope.
+    sequence_matches: MatchResults = collections.defaultdict(list)
+
+    sequence: collections.deque[FeatureSet] = collections.deque(maxlen=SEQUENCE_SIZE)
+
     for ch in extractor.get_calls(ph, th):
-        ifeatures, imatches = find_call_capabilities(ruleset, extractor, ph, th, ch)
-        for feature, vas in ifeatures.items():
+        cfeatures, cmatches = find_call_capabilities(ruleset, extractor, ph, th, ch)
+        for feature, vas in cfeatures.items():
             features[feature].update(vas)
 
-        for rule_name, res in imatches.items():
+        for rule_name, res in cmatches.items():
             call_matches[rule_name].extend(res)
 
+        sequence.append(cfeatures)
+        sfeatures: FeatureSet = collections.defaultdict(set)
+        for call in sequence:
+            for feature, vas in call.items():
+                sfeatures[feature].update(vas)
```

Comment:
should there be specific handling for repeated calls (in a loop)?
not necessarily here, but something to consider in general

Reply:
we'd maybe want to consider using the call return address to identify such cases